src/freenet/node/NodeIPPortDetector.java

```diff
@@ -58,7 +58,7 @@ public void shouldUpdate(){
 	 */
 	FreenetInetAddress[] detectPrimaryIPAddress() {
 		FreenetInetAddress addr = crypto.getBindTo();
-		if(addr.isRealInternetAddress(false, true, false)) {
+		if(addr.isRealInternetAddress(false, true, true)) {
```

Comment:
TODO: add config option [ ] only allow operation as localhost, for example for tor or i2p.